GoogleForCreators/web-stories-wp
# Updated parseInput validation to take both min and max into account

## Context


Ensure that all `opacityInput`s are within the given range [0, 100] 

## Summary


Fixes an oversight in `input/validation.js`.  Takes into account the combination of the `min` and `max` options now.

## Relevant Technical Choices


`input/validation.js` was overwriting the return value incorrectly if the `min` and `max` options were both set instead of chaining the logic.

## To-do



## User-facing changes


Before/After:
https://user-images.githubusercontent.com/14057928/165819621-25f1e468-3fbf-46ae-8aa9-1500450280f5.mp4

## Testing Instructions






This PR can be tested by following these steps:

1. Go to the Story Editor
2. Add components that have an opacity field (Text, Shape, Sticker, Image, etc).
3. Update the opacity field with any numbers outside the range of [0, 100] and press Enter or click outside the field

On Staging:
- Notice the opacity field allows for negative numbers but doesn't allow for numbers greater that 100

On Branch:
- Notice the opacity field doesn't allow for negative numbers, it will set the field to 0 if a negative number is entered and still doesn't allow for numbers greater than 100


## Reviews

### Does this PR have a security-related impact?
No


### Does this PR change what data or activity we track or use?
No


### Does this PR have a legal-related impact?
No


## Checklist




---



Fixes #4100
